### DISCUSSION TITLE
Plugins in Rust/Wasm

### DISCUSSION THREAD

**Discussion by alensiljak** [beancount/beancount#763] [Show and tell]
https://github.com/orgs/ledger-rs/discussions/40

I'm gonna raise this topic also here as it might be interesting for the Beancount community as well.
I was just at the Rust meetup where I learnt about the plugin system used in a terminal multiplexer. But what is interesting, the plugins are in Wasm. So they are multiplatform by design. And they are utilised from a terminal that is compiled to a native library in Rust.
Wasi target offers various levels of protection for the plugin systems in general.
Definitely something I'm going to investigate into more detail.
The link to the discussion about this topic in general is given above, while here we can go in-depth with potential use in the Beancount ecosystem.

**Comment by airtonix** [comment]
Wasm is a good choice in general for locally controlled data, but we should consider https://github.com/extism/extism instead; purely because it means less work and you're not limited to rust.
